tests/testthat/test-utils.R

```diff
@@ -20,8 +21,118 @@ test_that("log scale [PMP-TEST-063]", {
 })
 
 test_that("rot_x and rot_y [PMP-TEST-064]", {
-  expect_is(p + rot_x(), "gg")
-  expect_is(p + rot_y(), "gg")
+  x <- p + rot_x(angle = 33)
+  expect_is(x, "gg")
+  expect_equal(x$theme$axis.text.x$angle, 33)
+
+  y <- p + rot_y(angle = 11)
+  expect_equal(y$theme$axis.text.y$angle, 11)
+
+  x <- p + rot_x(vertical = TRUE)
+  expect_equal(x$theme$axis.text.x$angle, 90)
+  expect_equal(x$theme$axis.text.x$hjust, 1)
+
+  y <- p + rot_y(vertical = TRUE)
+  expect_equal(y$theme$axis.text.y$angle, 90)
+  expect_equal(y$theme$axis.text.y$vjust, 1)
+
+  x1 <- p + rot_x(vertical = TRUE, hjust = "bottom")
+  expect_equal(x1$theme$axis.text.x$angle, 90)
+  expect_equal(x1$theme$axis.text.x$hjust, 0)
+
+  x2 <- p + rot_x(vertical = TRUE, hjust = "top")
+  expect_equal(x2$theme$axis.text.x$angle, 90)
+  expect_equal(x2$theme$axis.text.x$hjust, 1)
+
+  y1 <- p + rot_y(vertical = TRUE, vjust = "left")
+  expect_equal(y1$theme$axis.text.y$angle, 90)
+  expect_equal(y1$theme$axis.text.y$vjust, 1)
+
+  y2 <- p + rot_y(vertical = TRUE, vjust = "right")
+  expect_equal(y2$theme$axis.text.y$angle, 90)
+  expect_equal(y2$theme$axis.text.y$vjust, 0)
+})
+
+test_that("rot_at - rotate list of plots", {
+  etas <- paste0("ETA", 1:3)
+  co <- c("STUDYc", "CPc", "RF")
+
+  x0 <- eta_cat(id, x = co, y = etas)
+
+  expect_identical(
+    names(x0)[1:3],
+    c("ETA1vSTUDYc", "ETA2vSTUDYc", "ETA3vSTUDYc")
+  )
+
+  x <- rot_at(x0, at = "ETA2vSTUDYc", angle = 11)
+  expect_is(x, "list")
+  expect_is(x[[1]], "gg")
+  expect_equal(x$ETA2vSTUDYc$theme$axis.text.x$angle, 11)
+  expect_null(x$ETA1vSTUDYc$theme$axis.text.x$angle)
+
+  x <- rot_at(x0, at = c("ETA2vSTUDYc", "ETA1vRF"), angle = 11)
+  expect_is(x, "list")
+  expect_equal(x$ETA2vSTUDYc$theme$axis.text.x$angle, 11)
+  expect_equal(x$ETA1vRF$theme$axis.text.x$angle, 11)
+
+  x <- rot_at(x0, re = "ETA2", angle = 33)
+  expect_is(x, "list")
+  expect_is(x[[1]], "gg")
+  expect_equal(x$ETA2vSTUDYc$theme$axis.text.x$angle, 33)
+  expect_equal(x$ETA2vCPc$theme$axis.text.x$angle, 33)
+  expect_equal(x$ETA2vRF$theme$axis.text.x$angle, 33)
+  expect_null(x$ETA1vSTUDYc$theme$axis.text.x$angle)
+  expect_null(x$ETA3vCPc$theme$axis.text.x$angle)
+
+  expect_error(rot_at(x0, at = "kyle"), "requested names not found")
+  expect_warning(rot_at(x0, re = "banana"), "did not find any plots")
+  expect_error(rot_at(x0[[1]]), "must be a list of gg")
+  expect_error(rot_at(unname(x0)), "must be named")
+})
+
+test_that("rot_xy", {
+  g1 <- dv_pred(data)
+  g2 <- dv_ipred(data)
+  p1 <- g1 + g2
+  p2 <- g1/g2
+  l <- list(a = g1, b = g2)
+  lp <- list(a = p1, b = p2)
+  l0 <- unname(l)
+
+  # rotate gg
+  a <- rot_xy(g1)
+  b <- g1 + rot_x()
+  expect_equal(a, b)
+
+  # rotate patchwork
+  a <- rot_xy(p1)
+  b <- p1 & rot_x()
+  expect_equal(a, b)
+
+  # list of gg
+  a <- lapply(l, rot_xy)
+  expect_is(a, "list")
+  if(R.version$minor >= 4.1) {
```

Comment:
Getting test failure here, only on CI for earlier R versions. The plots look as expected to me and I don't know whats the difference in `plot_env` for the earlier versions. I'm going to skip for now so this doesn't hold things up. 

```
 ══ Failed ══════════════════════════════════════════════════════════════════════
  ── 1. Failure ('test-utils.R:105:3'): rot_xy ───────────────────────────────────
  `a` not equal to `b`.
  Component "plot_env": Component "...": target is not list-like
  
  ── 2. Failure ('test-utils.R:110:3'): rot_xy ───────────────────────────────────
  `a` not equal to `b`.
  Component "plot_env": Component "...": target is not list-like
  Component "patches": Component "plots": Component 1: Component 9: Component 1: target is not list-like
  ```